java/semantickernel-api/src/main/java/com/microsoft/semantickernel/memory/MemoryRecordMetadata.java

```diff
@@ -79,13 +87,14 @@ public String getAdditionalMetadata() {
      * @param externalSourceName Name of the external source if isReference is true.
      * @param additionalMetadata Field for saving custom metadata with a memory.
      */
+    @JsonCreator
     public MemoryRecordMetadata(
-            boolean isReference,
-            @Nonnull String id,
-            @Nonnull String text,
-            @Nonnull String description,
-            @Nonnull String externalSourceName,
-            @Nonnull String additionalMetadata) {
+            @JsonProperty("reference") boolean isReference,
```

Comment:
These need to match what's in dotnet/src/SemanticKernel.Abstractions/Memory/MemoryRecordMetadata.cs